### DISCUSSION TITLE
#17 - Retro

### DISCUSSION THREAD

**Discussion by Annalytic-programming** [CUMGroup/AMOGUS#141] [Blog]
# How you doin’ AMOGUS friends?

Last week we finished developing our project as good as possible. A few small things are still being implemented, such as refactoring or bugfixing, but the vast majority is complete. So it was on: The final Retro! And we would also like to give you a few insights into it.

To loosen things up, we just wrote down a few terms that came to mind for the project. 
![](https://github.com/CUMGroup/AMOGUS/assets/97878953/65ec2653-b814-49e4-a0d7-5eb26424f8c4)

after that we have the classic enumeration of:
... that went well
... that didn't go well
... Actions
... ideas
done in a slightly modified form:
![](https://github.com/CUMGroup/AMOGUS/assets/97878953/abcb776d-32d1-4ccc-a0f1-13cc82458c5c)

But lets talk about the single points a bit more:

## What worked well
- **Task assignment:** That went very well for us. at the beginning of each sprint, everyone took on exactly the tasks that he or she wanted and was able to do.
- **Teamwork:** I don't think I need to say any more than that. the teamwork was great thank you for that @AlexMi-Ha @jas20202 @WachtelHD @elnaz-gharoon 
- **Speeding up in semester 4:** we are particularly proud of this: after being slightly behind schedule in semester 3, we really stepped up our game again in semester 4
- **Help each other:** That goes hand in hand with teamwork
- **Communication (physical presence):** in presence our communication was great. online it was rather meager
- **The CUMGroup Team Spirit:** We have even taken this team spirit so far that we are now working on our own small website for the CUMGroup
- **Alex:** @AlexMi-Ha was once again a carry in our team. he did inhuman things again this time.

## What doesn't work well
- **Time & Task tracking:** that was a problem that came up more often. times were often not tracked or only tracked sporadically
- **PR reviews:** Our PRs sometimes needed a very long time to be reviewed.
- **atomic PR's:** Wir hatten zu viele PRs die zu überladen waren
- **Loss of a Teammember:** Unfortunately we had to part with a team member in the course of the 4th semester
- **Nicks laptop and sharing screen (& his localhost):** If the technology doesn't want to play along (or apple)
![](https://tenor.com/de/view/ph-apple-ifag-iphone-ip10-gif-13174434)
- **Communication (online):** Our online communication was very poor. Messages were not answered or answered late
- **PR refactoring (after requestet changes):** Even when PRs were answered quickly, requested changes were slow to implement
## What we need to do better next time
- **Structure / Vision:**  Unfortunately, we quickly lost sight of the vision and we didn't have a clear goal in mind
- **Time management:** Our time management could be improved. we took on too much for the time we had.
- **Answer to messages (communication):** our online communication in particular could be improved.
- **Little sprint retros (weeklies):** It would have been beneficial if we had weekly recaps
- **PRs:** PRs should be answered and processed faster in the future

## What else We want to say
- **Wambo:** i wambo; you wambo; he, she, it, wambo; the wambology
- **AMOGUS**
- **stay sus**
- **CUMGroup**
- **That is fine**
- **Thank you:** We not only want to thank all our team members, but also you, the ones who commented on our posts, the whole DHBW community

Even if we had our difficulties and problems, we are still proud of AMOGUS. We were able to create a good project and learned a lot from it.

Now only the presentation follows, which we hope to be able to complete with flying colors.

Thanks to you all!

Yours truly 
CUMGroup (AMOGUS)

**Comment by Luca-Hackl** [comment]
It's truly fantastic to read such a comprehensive and reflective retrospective from your team, CUMGroup. It's clear from your post that you all put a great deal of thought and effort into your project, AMOGUS, and that despite the challenges, you've come away with a stronger sense of teamwork and valuable lessons for future projects.

Your honesty in examining both your strengths and weaknesses demonstrates a high level of maturity and dedication. It's very encouraging to see that you have not only identified the areas where you did well but also recognized the areas where there is room for improvement. 

The commitment to task assignment, teamwork, and supporting each other is particularly commendable. It shows a group of individuals who are not only skilled in their individual tasks but also deeply committed to each other's success.

As you've rightly pointed out, there are also some areas for improvement, like time management and online communication. It's important to remember that these are common challenges in many teams and are a part of the learning process. I'm confident that with the self-awareness and dedication you've demonstrated in this retrospective, you'll successfully address these issues in your future endeavors.

I'm looking forward to seeing your team continue to grow and excel in the future. Your journey so far is truly inspiring and demonstrates the power of teamwork, perseverance, and continuous learning. Best of luck with your presentation and future projects!

Keep up the excellent work!

Best Regards, 
RALF

**Reply by AlexMi-Ha** [reply]
Dear RALF,
Thank you for your fantastic feedback on our retrospective. We appreciate your support and recognition of our efforts. Your encouragement means a lot to us as we continue to improve and grow. Looking forward to sharing our future progress with you.

Yours truly,
CUMGroup

**Comment by vans1646** [comment]
Dear CUMGroup,
an era is coming to an end.
It was great to read all of your blog posts and considering some of your retro comments for you it was also a ride. 
I really enjoyed to read all the details about your progress but also the challenges you had to encounter. Especially that you've lost a team member was new to me and was probably also a big challenge for you since you had to carry as a team the new workload BUT you did it while also doing a great job within your project.

Concerning some of your retro points, I think that it would've really been good if you would have done a weekly retro. Especially in the beginning it was really useful to our team because there were many things we had to change quickly for a better workflow. We also struggled with online responses and one member who was really affected by us not answering mentioned that in on of our retros and since then we all really watched out to answer in time or to communicate better under our tickets. Besides that it is really helpful it's also the normal SCRUM routine so perhaps there's really something behind the theory ;)
The final retro you did is there to see which lessons were learned and what you could do differently and I think you did that really well.

Perhaps you will write one last blog for the project in which you give a recap about all the cool features you managed to implement, I'd be glad to read!

Cheers,
Capyclue

**Comment by eileen522** [comment]
Congratulations, CUMGroup, on completing your project!
It's great to hear that you made significant progress and that the vast majority of the work is complete. Task assignment and teamwork seem to have been key strengths, with everyone taking on tasks they wanted and working well together. It's always inspiring to see a team come together and achieve their goals.

We can understand the challenges you faced with time and task tracking, as well as PR reviews taking longer than expected. These are common issues that teams encounter, but they can be addressed with better processes and communication. Similarly, the loss of a team member can be difficult, but it's important to adapt and continue moving forward.

Improving communication, both online and in terms of answering messages, is a valuable area for growth. Regular sprint retrospectives can help identify areas for improvement and keep the team aligned. It's also crucial to address PRs promptly and implement requested changes in a timely manner.

Maintaining a clear vision and managing time effectively are vital aspects of successful project management. Setting clear goals and regularly reviewing progress against them can help ensure that everyone stays on track. And, of course, a positive team spirit and gratitude towards fellow team members and the wider community are commendable.

As you move forward with your presentation, we wish you all the best. It sounds like you have learned a lot from this project, and that's a valuable outcome in itself. Keep up the good work and continue striving for excellence. Congratulations again on your achievements, and thank you for sharing your journey with us!

- eiCompany
